shared/utility/support/yaml_log_level.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+#ifndef UTILITY_SUPPORT_YAML_LOG_LEVEL_HPP
+#define UTILITY_SUPPORT_YAML_LOG_LEVEL_HPP
+
+#include <fmt/format.h>
+#include <map>
+#include <nuclear>
+#include <string>
+#include <yaml-cpp/yaml.h>
+
+namespace YAML {
+
+template <>
+struct convert<::NUClear::LogLevel> {
+    static Node encode(const ::NUClear::LogLevel& rhs) {
+        Node node;
+
+        switch (rhs) {
+            case NUClear::TRACE: node = "TRACE"; break;
+            case NUClear::DEBUG: node = "DEBUG"; break;
+            case NUClear::INFO: node = "INFO"; break;
+            case NUClear::WARN: node = "WARN"; break;
+            case NUClear::ERROR: node = "ERROR"; break;
+            case NUClear::FATAL: node = "FATAL"; break;
+            default: node = "DEBUG"; break;
+        }
+
+        return node;
+    }
+
```

Comment:
This is a simple function to read the source for, but I think we should still be adding @brief etc. to stuff like this, even if it's only for the day of auto generated doxygen-tation